Packet++/header/SomeIpLayer.h

```diff
@@ -0,0 +1,361 @@
+#ifndef PACKETPP_SOMEIP_LAYER
+#define PACKETPP_SOMEIP_LAYER
+
+#include "Layer.h"
+#include <unordered_set>
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+/**
+ * @class SomeIpLayer
+ * Represents a SOME/IP protocol layer
+ */
+class SomeIpLayer : public Layer
+{
+public:
+	/**
+	 * SOME/IP message types
+	 */
+	enum class MsgType : int

```

Comment:
nit: since message type is 1-byte I think it can be `enum class MsgType : uint8_t`